ts/redux/async_dispatcher.ts

```diff
@@ -58,15 +59,47 @@ export const asyncDispatcher = {
             const provider = web3Wrapper.getProvider();
             const contractAddresses = getContractAddressesForNetworkOrThrow(networkId as number);
             const zrxTokenContract = new ERC20TokenContract(contractAddresses.zrxToken, provider);
-            const [ethBalanceInWei, zrxBalance] = await Promise.all([
+            const [ethBalanceInWei, zrxBalance, zrxAllowance] = await Promise.all([
                 web3Wrapper.getBalanceInWeiAsync(address),
                 zrxTokenContract.balanceOf.callAsync(address),
+                zrxTokenContract.allowance.callAsync(address, contractAddresses.erc20Proxy),
             ]);
+
             dispatcher.updateAccountEthBalance({ address, ethBalanceInWei });
             dispatcher.updateAccountZrxBalance(zrxBalance);
+            dispatcher.updateAccountZrxAllowance(zrxAllowance);
         } catch (e) {
             logUtils.warn(e);
             return;
         }
     },
+
+    increaseZrxAllowanceAndDispatchToStoreIfNeededAsync: async (
+        providerState: ProviderState,
+        networkId: Network,
+        amountToStakeBaseUnits: BigNumber,
+        dispatcher: Dispatcher,
+    ) => {
+        const { provider } = providerState;
+        const ownerAddress = (providerState.account as AccountReady).address;
+        const gasInfo = await backendClient.getGasInfoAsync();
+        const gasPriceInGwei = new BigNumber(gasInfo.fast / 10);
+        const gasPriceInWei = gasPriceInGwei.multipliedBy(1000000000);
```

Comment:
I know this is in `blockchain.ts` but seems useful.. maybe it should be another method on backendClient. Also just for reference here is similar code in instant https://github.com/0xProject/0x-monorepo/blob/3973dec9944a10c62ec6c3d15d973a00a6ea45a6/packages/instant/src/util/gas_price_estimator.ts#L31
The `estimatedTimeMs` should be useful for the countdown.